core/domain/learner_goals_services_test.py

```diff
@@ -256,3 +258,63 @@ def test_get_all_topic_ids_in_learn(self) -> None:
         self.assertEqual(
             learner_goals_services.get_all_topic_ids_to_learn(
                 self.viewer_id), [self.TOPIC_ID_1, self.TOPIC_ID_2])
+
+    def test_remove_topics_from_learn_goal_executed_correctly(
+        self
+    ) -> None:
+        self.assertEqual(self._get_all_topic_ids_to_learn(
+            self.viewer_id), [])
+
+        # Add topics to learner goals.
+        learner_goals_services.mark_topic_to_learn(
+            self.viewer_id, self.TOPIC_ID_1)
+        learner_goals_services.mark_topic_to_learn(
+            self.viewer_id, self.TOPIC_ID_2)
+        self.assertEqual(self._get_all_topic_ids_to_learn(
+            self.viewer_id), [self.TOPIC_ID_1, self.TOPIC_ID_2])
+
+        # Remove a topic from learner goals.
+        learner_goals_services.remove_topics_from_learn_goal(
+            self.viewer_id, [self.TOPIC_ID_1])
+        self.assertEqual(self._get_all_topic_ids_to_learn(
+            self.viewer_id), [self.TOPIC_ID_2])
+
+        # Check if the removed topic is not present in the learner goals.
+        learner_goals = learner_goals_services.get_all_topic_ids_to_learn(
+            self.viewer_id)
+        self.assertNotIn(self.TOPIC_ID_1, learner_goals)
+
+        # Check if the remaining topic is still present in the learner goals.
+        self.assertIn(self.TOPIC_ID_2, learner_goals)
+
+        # Remove the remaining topic from learner goals.
+        learner_goals_services.remove_topics_from_learn_goal(
+            self.viewer_id, [self.TOPIC_ID_2])
+        self.assertEqual(self._get_all_topic_ids_to_learn(
+            self.viewer_id), [])
+
+        # Check if both topics are not present in the learner goals.
+        learner_goals = learner_goals_services.get_all_topic_ids_to_learn(
+            self.viewer_id)
+        self.assertNotIn(self.TOPIC_ID_1, learner_goals)
+        self.assertNotIn(self.TOPIC_ID_2, learner_goals)
+
+    def test_remove_topics_when_learner_goals_model_does_not_exist(
+        self
+    ) -> None:
+        non_existent_user_id = 'non_existent_user_id'
+        self.assertIsNone(
+            user_models.LearnerGoalsModel.get(
+            non_existent_user_id, strict=False
+            )
+        )
+
+        # Ensure that the function does not raise any error when the
+        # learner_goals_model does not exist for the given user_id.
+        with mock.patch(
```

Comment:
I don't understand why we need to use a mock here -- or at least I don't understand what it has to do with the previous comment. If the point is to show that the function doesn't raise an error, then why not just call the function directly and ensure it doesn't error? (Or is the comment wrong?)

Reply:
Done!